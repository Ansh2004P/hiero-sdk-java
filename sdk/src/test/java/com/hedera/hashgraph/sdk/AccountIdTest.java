package com.hedera.hashgraph.sdk;

import com.google.protobuf.InvalidProtocolBufferException;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

class AccountIdTest {
    @Test
    @DisplayName("using toBytes and fromBytes will produce the correct Id")
    void keyGenerates() throws InvalidProtocolBufferException {
        AccountId id = AccountId.fromString("0.0.5005");

        assertNotNull(id);
        assertNotNull(id.toBytes());

        byte[] idBytes = id.toBytes();
        AccountId newId = AccountId.fromBytes(idBytes);

        assertEquals(id, newId);
    }


}
