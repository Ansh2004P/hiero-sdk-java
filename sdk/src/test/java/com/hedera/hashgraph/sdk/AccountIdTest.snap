com.hedera.hashgraph.sdk.AccountIdTest.fromBytes=[
  "0.0.5005"
]


com.hedera.hashgraph.sdk.AccountIdTest.fromSolidityAddress=[
  "0.0.5005"
]


com.hedera.hashgraph.sdk.AccountIdTest.fromSolidityAddressWith0x=[
  "0.0.5005"
]


com.hedera.hashgraph.sdk.AccountIdTest.fromString=[
  "0.0.5005"
]


com.hedera.hashgraph.sdk.AccountIdTest.toBytes=[
  "188d27"
]


com.hedera.hashgraph.sdk.AccountIdTest.toSolidityAddress=[
  "000000000000000000000000000000000000138d"
]